qa/tasks/cephfs/test_mds_metrics.py

```diff
@@ -511,3 +511,67 @@ def test_client_metrics_and_metadata(self):
             if not (client_metadata[i]['valid_metrics']):
                 raise RuntimeError("valid_metrics not found!")
 
+    def test_metrics_for_filtered_mds_rank(self):
+        # validate
+        valid, metrics = self._get_metrics(
+            self.verify_mds_metrics(client_count=TestMDSMetrics.CLIENTS_REQUIRED), 30)
+        log.debug("metrics={0}".format(metrics))
+        self.assertTrue(valid)
+
+        # grow the mds cluster
+        self.fs.grow(2)
+
+        fscid = self.fs.id
+        # spread directory per rank
+        self._spread_directory_on_all_ranks(fscid)
+
+        # spread some I/O
+        self._do_spread_io(fscid)
+
+        # wait a bit for mgr to get updated metrics
+        time.sleep(5)
+
+        # validate
+        valid, metrics = self._get_metrics(
+            self.verify_mds_metrics(active_mds_count=2, client_count=TestMDSMetrics.CLIENTS_REQUIRED), 30)
+        log.debug("metrics={0}".format(metrics))
+        self.assertTrue(valid)
+
+        # initiate a new query with `--mds_rank` filter and validate if
+        # we get metrics *only* from that mds.
+        filtered_mds = 1
+        valid, metrics = self._get_metrics(
+            self.verify_mds_metrics(client_count=TestMDSMetrics.CLIENTS_REQUIRED,
+                                    ranks=[filtered_mds]), 30, '--mds_rank={}'.format(filtered_mds))
+        log.debug("metrics={0}".format(metrics))
+        self.assertTrue(valid)
+
+        if len(metrics['client_metadata']) != TestMDSMetrics.CLIENTS_REQUIRED or\
+               len(metrics['global_metrics']) != TestMDSMetrics.CLIENTS_REQUIRED or\
+                   (len(metrics['metrics']) != 2 and metrics['metrics'][1] != f"mds.{filtered_mds}"):
+                   raise RuntimeError("Incorrect output for filtered mds_rank")
```

Comment:
This is excellent, but please drop it. 

My bad, I've checked `verify_mds_metrics()` again now. This testing is already done in `verify_mds_metrics()` via `test_query_mds_filter()`. Look at [this](https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mds_metrics.py#L43). 

> Or may be I would've miss something in test_query_mds_filter.

Yes, that's right. `test_query_mds_filter` does the necessary tests for the filtered metrics.